backend-db/Rhyolite/Backend/DB/Serializable.hs

```diff
@@ -42,7 +45,7 @@ import qualified Control.Monad.State as S
 -- It "disallows" (makes harder) arbitrary IO.
 -- It "disallows" (makes harder) catching IO exceptions *inside* the transaction.
 newtype Serializable a = Serializable (ReaderT Pg.Connection (LoggingT IO) a)
-  deriving (Functor, Applicative, Monad, MonadThrow, MonadLogger)
+  deriving (Functor, Applicative, Monad, MonadCatch.MonadThrow, MonadLogger)
   -- NOTE: We *intentionally* leave out
   --   - 'MonadCatch' so you can't accidentally mask a serialization error from the outer retry logic.
```

Comment:
um

Reply:
haha that's confusing.... I'll import `MonadThrow` unqualified.

Reply:
Rather than leaving out the typeclass, wouldn't it be better to provide a custom implementation that gives the behavior we want?

Reply:
What behavior would that be? A `catch` that just catches `SqlSerializationError` first and rethrows it before running your handler? I suppose that would ensure this error is never masked.

Reply:
I think that wouldn't be *technically* law-abiding

```haskell
catch (throwM e) f = f e
```

Presumably this law is for all `e` and `f`. If we "unmasked" `SqlSerializationError` in `catch` then any `f :: SqlSerializationError -> m a` would not result in `f e` but an exception.

How much do we care...

Reply:
FTR:

```haskell
instance MonadCatch Serializable where
  catch m f = MonadCatch.catches m
    [ MonadCatch.Handler $ \(e :: SqlSerializationError) -> MonadCatch.throwM e -- Never let a handler mask this exception
    , MonadCatch.Handler f
    ]
````

Reply:
we could make it (kinda) law-abiding again by also changing our `throwM` i think.  it would be tricky and almost certainly not worthwhile for what it actually buys.  I'm just not sure we actually want `MonadCatch` here though, particularly since also any other `SqlError` could cause the transaction to abort and would need to be handled specially if you don't abort the `Serializable` action also ...